sat/meshed.go

```diff
@@ -83,7 +93,7 @@ func (s *Sat) handleFnResult(msg grav.Message, result interface{}, fnErr error)
 		return
 	}
 
-	if err := seq.HandleStepErrs([]sequence.FnResult{*fnr}, step.Exec); err != nil {
+	if err = seq.HandleStepErrs([]sequence.FnResult{*fnr}, step.Exec); err != nil {
```

Comment:
Since if statements have their own scope and as such these variables are actually truly new instantiations, I actually don't think I agree with removing the `:` here.

Reply:
This still counts as shadowing though, and Goland paints it green (to mark shadowed variables). Because it's an error, and we keep reassigning the value that the `err` variable holds, and the type is always the same, we can save both computer resources, and human thinking on redeclaration. I like to be precise and purposeful with how Go works, and not do things that aren't necessary and won't cause confusion.

In this case, redeclaring (`:=`) is confusing to me because I'm thinking: "hey, is there a mistake here that we need to redeclare this? Is something missing?", whereas a reassignment (`=`) communicates the intent that "yep, we know it's the same thing, no need for a different scope".

I whipped up a playground, one specific to errors and reassignment / redeclaration: https://go.dev/play/p/rL6y1UGH5yR

The only time I think there would be a difference / importance in redeclaration / reassignment is if we wanted to keep using the value of the err variable before an if-scoped redeclaration took place, but in that case I would opt to rename the if-scoped err variable to something different to catch the (very hard to find) bug due to shadowing.